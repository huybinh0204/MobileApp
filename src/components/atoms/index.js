export { default as AccountNumber } from './AccountNumber/AccountNumber';
export { default as Badge } from './Badge/Badge';
export { default as Balance } from './Balance/Balance';
export { default as BaseFormInput } from './BaseFormInput/BaseFormInput';
export { default as BaseText } from './BaseText/BaseText';
export { default as BottomSheet } from './BottomSheet/BottomSheet';
export { default as Card } from './Card/Card';
export { default as CurrencyInput } from './CurrencyInput/CurrencyInput';
export { default as Dash } from './Dash/Dash';
export { default as Form } from './Form/Form';
export { default as FormInput } from './FormInput/FormInput';
export { default as FormSubmitButton } from './FormSubmitButton/FormSubmitButton';
export { default as GeneralContainer } from './GeneralContainer/GeneralContainer';
export { default as IconSvg } from './IconSvg/IconSvg';
export { default as Link } from './Link/Link';
export { default as MainButton } from './MainButton/MainButton';
export { default as Money } from './Money/Money';
export { default as PdfView } from './PdfView/PdfView';
export { default as Plaid } from './Plaid/Plaid';
export { default as PressableListItem } from './PressableListItem/PressableListItem';
export { default as ProgressBar } from './ProgressBar/ProgressBar';
export { default as SegmentedTextInput } from './SegmentedTextInput/SegmentedTextInput';
export { default as SelectButton } from './SelectButton/SelectButton';
export { default as Switch } from './Switch/Switch';
export { default as TabItem } from './TabItem/TabItem';
export { default as Toast } from './Toast/Toast';
export { default as TransactionItem } from './TransactionItem/TransactionItem';
