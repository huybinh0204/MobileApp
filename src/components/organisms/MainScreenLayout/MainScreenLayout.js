import PropTypes from 'prop-types';
import React from 'react';
import { Platform } from 'react-native';
import { MainHeader } from '_components/molecules';
import { Container, Content } from './MainScreenLayout.styles';

const MainScreenLayout = ({
  backgroundColor,
  children,
  showBackButton,
  showMenuButton,
  testID,
  title,
}) => (
  <Container backgroundColor={backgroundColor} testID={testID}>
    <MainHeader showBackButton={showBackButton} showMenuButton={showMenuButton} title={title} />
    <Content enabled={Platform.OS === 'ios'} behavior="padding">
      {children}
    </Content>
  </Container>
);

MainScreenLayout.defaultProps = {
  backgroundColor: null,
  showBackButton: false,
  showMenuButton: true,
  title: null,
};

MainScreenLayout.propTypes = {
  backgroundColor: PropTypes.string,
  children: PropTypes.node.isRequired,
  showBackButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
  testID: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default MainScreenLayout;
