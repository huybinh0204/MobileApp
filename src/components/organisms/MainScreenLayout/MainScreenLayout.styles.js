import styled from '@emotion/native';

export const Container = styled.SafeAreaView(({ backgroundColor, theme }) => ({
  flex: 1,
  backgroundColor: backgroundColor ?? theme.colors.background,
}));

export const Content = styled.KeyboardAvoidingView({
  flex: 1,
});
