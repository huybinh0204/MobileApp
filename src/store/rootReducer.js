import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
