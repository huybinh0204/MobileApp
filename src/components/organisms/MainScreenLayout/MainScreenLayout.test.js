import { render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { createStore } from 'redux';
import { BaseText } from '_components/atoms';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import MainScreenLayout from './MainScreenLayout';

const Component = () => (
  <MainScreenLayout title="test" testID="test">
    <BaseText>test</BaseText>
  </MainScreenLayout>
);

describe('MainScreenLayout', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = createStore(rootReducer, {});
  });

  describe('structure', () => {
    it('should match the snapshot', async () => {
      const { toJSON } = render(<ComponentWithProviders component={Component} store={mockStore} />);

      await waitFor(() => {
        expect(toJSON()).toMatchSnapshot();
      });
    });
  });
});
