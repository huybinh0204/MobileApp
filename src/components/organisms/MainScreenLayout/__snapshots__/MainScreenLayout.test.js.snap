// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreenLayout structure should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="test"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="MenuButton"
                          >
                            <SvgMock
                              height={54.5}
                              width={54.5}
                            />
                          </View>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#1F1D25",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 29,
                                  "lineHeight": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            test
                          </Text>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                          >
                            test
                          </Text>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
